src/test/java/com/wikia/webdriver/common/core/configuration/Configuration.java

```diff
@@ -1,50 +1,66 @@
-import java.io.InputStream;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.commons.lang.StringUtils;
+import org.yaml.snakeyaml.Yaml;
+
+import com.wikia.webdriver.common.core.exceptions.TestEnvInitFailedException;
+import com.wikia.webdriver.common.properties.Credentials;
 
 /**
  * Configuration handler. This Class should handle run configuration and global properties.
- * Configuration handling: 1. If there is a System Property specified - return value of this
- * property 2. If no System Property is found - value is provided from configuration files
- * (config_sample.yml if no config.yml provided) NOTE: system property should override ANY property
- * specified in config files
- */
+ * Configuration handling:
+ * <ol>
+ * <li>Look for the property key in testConfig map, if key is present, return the value</li>
+ * <li>Look for the property key in system properties - return value of this property if key present
+ * </li>
+ * <li>If no System Property is found - value is provided from configuration files
+ * (config_default.yml and config.yml). Values provided in config.yml, are overriding values from
+ * config_default.yml</li>
+ * </ol>
+ * */
 public class Configuration {
 
-  private static Map<String, String> config;
+  private static final String DEFAULT_CONFIG_FILE_NAME = "config_default.yml";
+  private static final String LOCAL_CONFIG_FILE_NAME = "config.yml";
+  private static final Logger LOGGER = Logger.getLogger(Configuration.class.getName());
+
+  private static Map<String, String> defaultConfig;
+  private static Map<String, String> localConfig = new HashMap<>();
   private static Map<String, String> testConfig = new HashMap<>();
 
-  private Configuration() {
-  }
+  private Configuration() {}
 
   private static Map<String, String> readConfiguration() {
-    if (config == null) {
+    if (defaultConfig == null) {
       Yaml yaml = new Yaml();
-      InputStream input = null;
+
       try {
-        input = new FileInputStream(new File("config.yml"));
-      } catch (FileNotFoundException ex) {
-        try {
-          input = new FileInputStream(new File("config_sample.yml"));
-        } catch (FileNotFoundException ex2) {
-          throw new TestEnvInitFailedException("CAN'T LOCATE CONFIG FILE");
-        }
+        defaultConfig =
+            (Map<String, String>) yaml
+                .load(new FileInputStream(new File(DEFAULT_CONFIG_FILE_NAME)));
+      } catch (FileNotFoundException e) {
+        throw new TestEnvInitFailedException(String.format("CANNOT FIND DEFAULT CONFIG FILE : %s",
+            DEFAULT_CONFIG_FILE_NAME), e);
+      }
+
+      try {
+        defaultConfig.putAll((Map<String, String>) yaml.load(new FileInputStream(new File(
+            LOCAL_CONFIG_FILE_NAME))));
+      } catch (FileNotFoundException e) {
+        LOGGER.log(Level.INFO, "local config file not found", e);
```

Comment:
If I don't have `config.yml` file I get in terminal:
```
INFO: local config file not found
java.io.FileNotFoundException: config.yml (No such file or directory)
	at java.io.FileInputStream.open(Native Method)
	at java.io.FileInputStream.<init>(FileInputStream.java:146)
	at com.wikia.webdriver.common.core.configuration.Configuration.readConfiguration(Configuration.java:56)
	at com.wikia.webdriver.common.core.configuration.Configuration.getPropertyFromFile(Configuration.java:67)
	at com.wikia.webdriver.common.core.configuration.Configuration.getProp(Configuration.java:73)
	at com.wikia.webdriver.common.core.configuration.Configuration.getCredentialsFilePath(Configuration.java:97)
	at com.wikia.webdriver.common.core.annotations.User.<init>(User.java:26)
	at com.wikia.webdriver.common.core.annotations.User.<clinit>(User.java:12)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at java.lang.Class.getEnumConstantsShared(Class.java:3128)
	at java.lang.Class.enumConstantDirectory(Class.java:3149)
	at java.lang.Enum.valueOf(Enum.java:231)
	at sun.reflect.annotation.AnnotationParser.parseEnumValue(AnnotationParser.java:475)
	at sun.reflect.annotation.AnnotationParser.parseMemberValue(AnnotationParser.java:340)
	at java.lang.reflect.Method.getDefaultValue(Method.java:741)
	at sun.reflect.annotation.AnnotationType.<init>(AnnotationType.java:128)
	at sun.reflect.annotation.AnnotationType.getInstance(AnnotationType.java:85)
	at sun.reflect.annotation.AnnotationParser.parseAnnotation2(AnnotationParser.java:263)
	at sun.reflect.annotation.AnnotationParser.parseAnnotations2(AnnotationParser.java:117)
	at sun.reflect.annotation.AnnotationParser.parseAnnotations(AnnotationParser.java:70)
	at java.lang.reflect.Method.declaredAnnotations(Method.java:714)
	at java.lang.reflect.Method.getAnnotation(Method.java:700)
	at org.testng.internal.annotations.JDK15AnnotationFinder.findAnnotation(JDK15AnnotationFinder.java:107)
	at org.testng.internal.TestNGClassFinder.<init>(TestNGClassFinder.java:68)
	at org.testng.TestRunner.initMethods(TestRunner.java:415)
	at org.testng.TestRunner.init(TestRunner.java:241)
	at org.testng.TestRunner.init(TestRunner.java:211)
	at org.testng.TestRunner.<init>(TestRunner.java:158)
	at org.testng.SuiteRunner$DefaultTestRunnerFactory.newTestRunner(SuiteRunner.java:545)
	at org.testng.SuiteRunner.init(SuiteRunner.java:159)
	at org.testng.SuiteRunner.<init>(SuiteRunner.java:114)
	at org.testng.TestNG.createSuiteRunner(TestNG.java:1260)
	at org.testng.TestNG.createSuiteRunners(TestNG.java:1247)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1101)
	at org.testng.TestNG.run(TestNG.java:1018)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:295)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:84)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:90)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
```
Can we reduce this info to:
```
INFO: local config file not found
```
?